tensorflow/python/ops/variables.py

```diff
@@ -226,7 +226,11 @@ def __init__(self,
           expected_shape=expected_shape)
 
   def __str__(self):
-    return str(self._snapshot)
+    return self.__repr_()
```

Comment:
Done!

As a note, I've read elsewhere that the difference between them is that `__repr__` is supposed to be more descriptive, and often times copy-pastable as python code, whereas `__str__` is meant to be more readable. In this case, it means that we could do:

```python
>>> variable.__str__()
<tf.Variable 'name' shape=(10, 2) dtype=float32>
>>>variable.__repr__()
tf.Variable()
```

The reason I didn't try doing this in the original version of the PR is because
a) I assume this would be a much bigger change than what the issue was asking for and
b) I found that this wasn't consistent within tf:

```python
>>> tf.float32.__str__()
<dtype: 'float32'>
>>> tf.float32.__repr__()
tf.float32

>>> tf.constant(0.1).__str__()
Tensor("Const:0", shape=(), dtype=float32)
>>> tf.constant(0.1).__repr__()
<tf.Tensor 'Const:0' shape=() dtype=float32>
```